streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableFilter.java

```diff
@@ -77,6 +77,7 @@ public void process(K key, Change<V> change) {
             V newValue = computeValue(key, change.newValue);
             V oldValue = sendOldValues ? computeValue(key, change.oldValue) : null;
 
+            if (oldValue == null && newValue == null) return; // unnecessary to forward here.
```

Comment:
I'm not sure if we can optimize the case when `sendOldValues` is false as well. Following your example:

KTable T1 = builder.table("source-topic");
KTable T2 = table.filter(value > 2);
T2.to("sink-topic");

And suppose the "source-topic" is piping the messages to T1 as: {a: 3}, {b: 5}, {a: 1}...

When {a: 3} is passed from T1 to T2, the filter will pass and hence it is forwarded to downstream operators already; so now when later {a: 1} is passed from T1 to T2, meaning "modifying the value with key {a} from 3 to 1", the filter will not pass any more, and hence in this case we need to forward a {a: null} record downstreams in order to indicate the previously forwarded {a: 3} has now been deleted in T2, right? Otherwise the sink topic will have the following messages:

{a: 3}, {b: 5}, ...

whereas the right sequence should be

{a: 3}, {b: 5}, {a: null}, ...